src/griddyn_shared/griddyn_export.h

```diff
@@ -84,41 +84,42 @@ extern "C"
     @param[in] objectType  the specific type of the component to create
     @return a gridDynObject that represents the newly created object
     */
-    GRIDDYN_EXPORT gridDynObject gridDynObject_create (const char *componentType, const char *objectType);
+    GRIDDYN_EXPORT gridDynObject *gridDynObject_create (const char *componentType, const char *objectType);
 
     /** clone an object from an existing object
     @details a new object is constructed that is a clone of the original object
     @param[in] obj the original object
     @return a gridDynObject that represents the newly created object
     */
-    GRIDDYN_EXPORT gridDynObject gridDynObject_clone (gridDynObject const obj);
+    GRIDDYN_EXPORT gridDynObject *gridDynObject_clone (gridDynObject const *obj);
```

Comment:
I don't think this correct.  griDynObject is typedef to void * so this becomes a const pointer to a void *  which I don't think is what we want

Reply:
ok I see you changed gridDynObject to be void, so this works but I am not sure this makes sense as it becomes much more complicated for users on the generate object since you now can't just use the type you always have to use with the *.   And I don't think this typical for C interfaces.

Reply:
I changed the typedef to void, these are now `gridDynObject const*` instead of `gridDynObject *const`

Reply:
Well, it's either this or we should drop the const altogether

I don't particularly like the pointer typedefs to begin with...

Reply:
lets drop all the const stuff for the opaque objects.  Playing around with it, they don't provide any additional protection, and they don't impact any wrappers for C++ around them.  So lets make it easy and drop all const around the opaque objects.  And have the opaque objects be a typedef to a void *